src/include/dhcp-server/dialogs.rb

```diff
@@ -378,27 +378,26 @@ def SectionTypeChoose
       par_id = @parent_id
       par_type = @parent_type
       while par_id != ""
-        parents = Builtins.add(parents, par_type)
+        parents << par_type
 
         par = DhcpServer.GetEntryParent(par_type, par_id)
         par_type = Ops.get(par, "type", "")
         par_id = Ops.get(par, "id", "")
       end
       possible = ["subnet", "host", "shared-network", "group", "pool", "class"]
-      if Builtins.contains(parents, "class") ||
-          Builtins.contains(parents, "host") ||
-          Builtins.contains(parents, "pool")
+      if ["class", "host", "pool"].include?(parents)
         return :back
       end
-      f = []
-      if Builtins.contains(parents, "subnet")
-        f = Builtins.add(f, "subnet")
-        f = Builtins.add(f, "shared-network")
+      filtered = []
```

Comment:
make sense, naming is always challenge...and previous `f` was horrible, it took me 5 minutes, just to recognize how it works, as old filer method and small `!` cause I think it is another allowed list.